### ISSUE TITLE
Custom storage type for every table.

### ISSUE THREAD

**Issue by killme2008** [GreptimeTeam/greptimedb#919]
### What problem does the new feature solve?

The GreptimeDB already supports storing data in the local file system, AWS S3 and Aliyun Oss, thanks to OpenDAL.

But this storage configuration is global. You can't have a table in the local file system and the other on S3.

I think we can make this configuration also present in table engine options. For example, you can create a table and specify its storage type:

```sql
create table monitor(CPU double, ts timestamp time index) with(regions=1, storage=s3);
```

If the SQL doesn't set the `storage` option, then using the global configuration.

### What does the feature do?

Describe as above.

**Comment by NiwakaDev** [comment]
@MichaelScofield 
Is this also blocked as #1709?

**Comment by MichaelScofield** [comment]
@NiwakaDev I don't think this issue is blocked by any other features, it can be resolved independently.

**Comment by NiwakaDev** [comment]
@killme2008 
This issue discussion already lands? If so, I'd like to work on this issue.

**Comment by killme2008** [comment]
@NiwakaDev We haven't discussed it in detail yet, but the syntax is as expected in the issue description.

Feel free to try it, we think it's helpful for users.

**Comment by NiwakaDev** [comment]
@killme2008 

Should we support this feature for file engine?: https://github.com/GreptimeTeam/greptimedb/blob/develop/src/datanode/src/datanode.rs#L487

**Comment by killme2008** [comment]
@NiwakaDev Do you mean to create a file engine table with custom storage?
It's already supported, we can create a file engine table with file in s3, see https://docs.greptime.com/reference/sql/create#s3
